### PR TITLE
vso-node-api version update

### PR DESCRIPTION
**Pull request by zxan1285**
## ✏️ Changes

- Build process changed to use `auth0-extensions-cli` like the rest of extensions.
- Updated `vso-node-api` dependency version - now using v6.5.0
  
## 📷 Screenshots
 
No visual changes
  
## 🔗 References
  
- Jira: https://auth0team.atlassian.net/browse/KEY-224

## 🎯 Testing

This change is already covered by unit tests and manual testing as it solely deals with build and dependency changes.

✅ This change has been tested in a Webtask
 
✅ This change has unit test coverage
  
🚫 This change has integration test coverage
  
🚫 This change has been tested for performance
  
## 🚀 Deployment
    
✅ This can be deployed any time
  
## 🎡 Rollout
  
In order to verify that the deployment was successful we will perform a smoke test in production before releasing it to customers.
  
## 🔥 Rollback

We will rollback if we notice any elevated failures calling the VSTS API
  
### 📄 Procedure
  
In the event of an issue, we will simply revert the version which will be rapid.
 
## 🖥 Appliance
  
**Note to reviewers:** ensure that this change is compatible with the Appliance.

### CODE REVIEW

**Comment by auth0extensions** [comment]
MD5: 1c9e99e48a495fe81d388fdb4900e59f

# NSP - Security Report

_This is an automated security audit of this project. Please do not modify its contents_

Vulnerabilities found: **28**
Affected Modules: **hoek, hoek, hoek, hoek, hoek, superagent, superagent, superagent, superagent, superagent, superagent, mime, mime, mime, mime, mime, superagent, superagent, superagent, uglify-js, uglify-js, uglify-js, uglify-js, constantinople, constantinople, lodash, lodash, deep-extend**
Summary:

- Prototype pollution attack
- Prototype pollution attack
- Prototype pollution attack
- Prototype pollution attack
- Prototype pollution attack
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Large gzip Denial of Service
- Incorrect Handling of Non-Boolean Comparisons During Minification
- Incorrect Handling of Non-Boolean Comparisons During Minification
- Regular Expression Denial of Service
- Regular Expression Denial of Service
- Sandbox Bypass Leading to Arbitrary Code Execution
- Sandbox Bypass Leading to Arbitrary Code Execution
- Prototype Pollution
- Prototype Pollution
- Prototype Pollution


## Detailed report
#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-extension-express-tools@1.1.8 > webtask-tools@3.3.0 > boom@2.10.1 > hoek@2.16.3**

---

#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-extension-tools@1.3.2 > webtask-tools@3.3.0 > boom@2.10.1 > hoek@2.16.3**

---

#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-source-control-extension-tools@2.7.2 > auth0-extension-tools@0.14.0 > jsonwebtoken@7.4.3 > joi@6.10.1 > hoek@2.16.3**

---

#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-visualstudio-deploy-extension@2.3.0 > jsonwebtoken@7.4.3 > joi@6.10.1 > hoek@2.16.3**

---

#### Prototype pollution attack

Affected Module: **hoek**
Installed version: **2.16.3**
Patched versions: **> 4.2.0 < 5.0.0 || >= 5.0.3**
Advisory: https://nodesecurity.io/advisories/566
Path: **auth0-visualstudio-deploy-extension@2.3.0 > webtask-tools@3.3.0 > boom@2.10.1 > hoek@2.16.3**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-extension-express-tools@1.1.8 > webtask-tools@3.3.0 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-extension-tools@1.3.2 > webtask-tools@3.3.0 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-oauth2-express@1.3.0 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-source-control-extension-tools@2.7.2 > auth0-oauth2-express@1.3.0 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-source-control-extension-tools@2.7.2 > auth0-extension-tools@0.14.0 > webtask-tools@2.2.0 > superagent@1.8.5**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **1.8.5**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-visualstudio-deploy-extension@2.3.0 > webtask-tools@3.3.0 > superagent@1.8.5**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-extension-express-tools@1.1.8 > webtask-tools@3.3.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-extension-tools@1.3.2 > webtask-tools@3.3.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-oauth2-express@1.3.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-source-control-extension-tools@2.7.2 > auth0-oauth2-express@1.3.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Regular Expression Denial of Service

Affected Module: **mime**
Installed version: **1.3.4**
Patched versions: **>= 1.4.1 < 2.0.0 || >= 2.0.3**
Advisory: https://nodesecurity.io/advisories/535
Path: **auth0-visualstudio-deploy-extension@2.3.0 > webtask-tools@3.3.0 > superagent@1.8.5 > mime@1.3.4**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-extension-express-tools@1.1.8 > auth0-extension-tools@1.3.3 > superagent@2.3.0**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-extension-tools@1.3.2 > superagent@2.3.0**

---

#### Large gzip Denial of Service

Affected Module: **superagent**
Installed version: **2.3.0**
Patched versions: **>=3.7.0**
Advisory: https://nodesecurity.io/advisories/479
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-source-control-extension-tools@2.7.2 > auth0-extension-tools@0.14.0 > superagent@2.3.0**

---

#### Incorrect Handling of Non-Boolean Comparisons During Minification

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>= 2.4.24**
Advisory: https://nodesecurity.io/advisories/39
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Incorrect Handling of Non-Boolean Comparisons During Minification

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>= 2.4.24**
Advisory: https://nodesecurity.io/advisories/39
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-source-control-extension-tools@2.7.2 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Regular Expression Denial of Service

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>=2.6.0**
Advisory: https://nodesecurity.io/advisories/48
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Regular Expression Denial of Service

Affected Module: **uglify-js**
Installed version: **2.2.5**
Patched versions: **>=2.6.0**
Advisory: https://nodesecurity.io/advisories/48
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-source-control-extension-tools@2.7.2 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > transformers@2.1.0 > uglify-js@2.2.5**

---

#### Sandbox Bypass Leading to Arbitrary Code Execution

Affected Module: **constantinople**
Installed version: **3.0.2**
Patched versions: **>=3.1.1**
Advisory: https://nodesecurity.io/advisories/568
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > constantinople@3.0.2**

---

#### Sandbox Bypass Leading to Arbitrary Code Execution

Affected Module: **constantinople**
Installed version: **3.0.2**
Patched versions: **>=3.1.1**
Advisory: https://nodesecurity.io/advisories/568
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-source-control-extension-tools@2.7.2 > auth0-oauth2-express@1.3.0 > jade@1.11.0 > constantinople@3.0.2**

---

#### Prototype Pollution

Affected Module: **lodash**
Installed version: **3.10.1**
Patched versions: **>=4.17.5**
Advisory: https://nodesecurity.io/advisories/577
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-oauth2-express@1.3.0 > auth0-api-jwt-rsa-validation@0.0.1 > lodash@3.10.1**

---

#### Prototype Pollution

Affected Module: **lodash**
Installed version: **3.10.1**
Patched versions: **>=4.17.5**
Advisory: https://nodesecurity.io/advisories/577
Path: **auth0-visualstudio-deploy-extension@2.3.0 > auth0-source-control-extension-tools@2.7.2 > auth0-oauth2-express@1.3.0 > auth0-api-jwt-rsa-validation@0.0.1 > lodash@3.10.1**

---

#### Prototype Pollution

Affected Module: **deep-extend**
Installed version: **0.4.2**
Patched versions: **>=0.5.1**
Advisory: https://nodesecurity.io/advisories/612
Path: **auth0-visualstudio-deploy-extension@2.3.0 > deep-extend@0.4.2**

---

File: build/webpack/server.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const logger = require('../../server/lib/logger');
 logger.info('Running development webpack server...');
 
 const options = {
+  disableHostCheck: false,
```
Comment (ghost) [inline build/webpack/server.js:10]: Shouldn't this be `true`?

**Review by ghost — CHANGES_REQUESTED** [review]
LGTM - just a small comment

File: build/webpack/server.js (hunk 1 above)
Comment (zxan1285) [reply build/webpack/server.js:10]: It is `false` by default, i'm disabling it for local testing to make it working with ngrok. IIRC, some time ago `disableHostCheck: true` was considered as security issue, even though it is using for local testing only.
So now i'm just enabling hostCheck back before committing changes.

**Review by ghost — APPROVED** [review]

**Pull request merged**